samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -152,6 +152,19 @@ public synchronized String registerProcessorAndGetId(final ProcessorData data) {
     return ephemeralPath;
   }
 
+  /**
+   * Deletes the ephemeral node of a processor in zookeeper.
+   * @param processorId uniqueId identifying the stream processor to delete.
+   */
+  public void deleteProcessorNode(String processorId) {
+    for (ProcessorNode processorNode : getAllProcessorNodes()) {
+      if (processorNode.getProcessorData().getProcessorId().equals(processorId)) {
```

Comment:
do u need to guard against NPE's anywhere here?

Reply:
`getAllProcessorNodes()` has sufficient guards for NPE and will not return null values for any fields . Adding the same checks here would be unnecessary.